streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -115,15 +116,18 @@ public StreamTask(final TaskId id,
         // to corresponding source nodes in the processor topology
         final Map<TopicPartition, RecordQueue> partitionQueues = new HashMap<>();
 
-        final TimestampExtractor timestampExtractor = config.getConfiguredInstance(StreamsConfig.TIMESTAMP_EXTRACTOR_CLASS_CONFIG, TimestampExtractor.class);
-
+        final TimestampExtractor deprecatedTimestampExtractor = config.getConfiguredInstance(StreamsConfig.TIMESTAMP_EXTRACTOR_CLASS_CONFIG, TimestampExtractor.class);
```

Comment:
Could we add a similar function in `StreamsConfig` as `keySerde` in which we capture the deprecated / new default values there so that we do not need to leak that logic here?